File: BulletinBoard/src/main/scala/EventManager.scala
class EventManager {
  var subscriptions = scala.collection.mutable.Map[String, List[String => Unit]]()

  def subscribe(event_type: String, handler: String => Unit): Unit = {
    if(subscriptions.contains(event_type))
      subscriptions(event_type) = subscriptions(event_type) :+ handler

    else
      subscriptions += (event_type -> List(handler))
  }

  def publish(event: List[String]): Unit = {
    val event_type = event(0)
    if(subscriptions.contains(event_type)){
      for(h <- subscriptions(event_type)){
        h(event(1))
      }
    }
  }
}

File: BulletinBoard_Java/src/bulletinboard/EventManager.java
package bulletinboard;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Arrays;

/**
 *
 * @author marcu
 */
public class EventManager {
    Map<String, List<Handler>> subscriptions;

    public EventManager() { 
         subscriptions = new HashMap<String, List<Handler>>();
    }
    
    
    
    public void subscribe(String eventType, Handler handler){
        subscriptions.computeIfAbsent(eventType, k -> new ArrayList<Handler>()).add(handler);
       
    }
    
    public void publish(List<String> event){
        String eventType = event.get(0);
        
        
        if(subscriptions.containsKey(eventType)){
            for(Handler handler : subscriptions.get(eventType))
                handler.handle(event);
        }
            
    }
    
}
